### PR TITLE
Auto-label PRs based on their content [skip-ci]

### PR DESCRIPTION
**Pull request by jolorunyomi**
This PR adds the GitHub action [PR Labeler](https://github.com/actions/labeler) to auto-label PRs based on their content. 

Labeling is managed with a configuration file `.github/labeler.yml` using the following [options](https://github.com/actions/labeler#usage).

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+cuDF (Java):
+  - java/
+  
+Ops:
+  - .github/
+  - /ci/
+  - conda/
+  - **/Dockerfile
+  - **/.dockerignore
+  - docker/
```
Comment (ajschmidt8) [inline .github/labeler.yml:25]: have these patterns been tested? I'm not sure that the file patterns in the `CODEOWNERS` file can be directly copied into this file. For instance, the [Common Examples](https://github.com/actions/labeler#common-examples) section says you would need to use `python/**/*` (not `python/`) to match any file (recursively) in the `python` directory. I think these will all need to be updated accordingly.

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]
Nice work on this. I think that the file/path patterns will need to be re-reviewed and adjusted according to the [actions/labeler README](https://github.com/actions/labeler) though.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by harrism** [comment]
This PR needs a description of what is in the PR.

**Comment by mike-wendt** [comment]
> This PR needs a description of what is in the PR.

@jolorunyomi can you update this PR and others with a description of what it is addressing? This way it can be auto-merged and saved in the update log?

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  - 'cpp/**'
+
+CMake:
+  - '**/CMakeLists.txt'
+  - '**/cmake/**'
+  
+cuDF (Java):
+  - 'java/**'
+  
+Ops:
```
Comment (mike-wendt) [inline .github/labeler.yml:19]: There doesn't seem to be an `Ops` label, I think we can remove this section

**Review by mike-wendt — CHANGES_REQUESTED** [review]

**Comment by mike-wendt** [comment]
@kkraus14 @harrism can you review this labeler config to suggest or add any other labels you would like to see automatically added for `cudf`?

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  - '.github/**'
+  - 'ci/**'
+  - 'conda/**'
```
Comment (kkraus14) [inline .github/labeler.yml:22]: We have a `conda` label it would be nice to put this under.

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+CMake:
+  - '**/CMakeLists.txt'
+  - '**/cmake/**'
+  
+cuDF (Java):
+  - 'java/**'
+  
+Ops:
+  - '.github/**'
+  - 'ci/**'
```
Comment (kkraus14) [inline .github/labeler.yml:21]: We also have a `gpuCI` label, but not sure if it makes sense to have that for this.

File: .github/labeler.yml (hunk 3 above)
Comment (mike-wendt) [reply .github/labeler.yml:22]: Agreed this should be it's own labeled section

File: .github/labeler.yml (hunk 4 above)
Comment (mike-wendt) [reply .github/labeler.yml:21]: Would be helpful to call this out automatically, we should create a label section just for the `ci` path

File: .github/labeler.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  - 'conda/**'
+  - '**/Dockerfile'
+  - '**/.dockerignore'
+  - 'docker/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:25]: These can be dropped

File: .github/labeler.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+CMake:
+  - '**/CMakeLists.txt'
+  - '**/cmake/**'
+  
+cuDF (Java):
+  - 'java/**'
+  
+Ops:
+  - '.github/**'
```
Comment (mike-wendt) [inline .github/labeler.yml:20]: I would drop these

**Review by kkraus14 — APPROVED** [review]
lgtm now

File: .github/labeler.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  - 'cpp/**'
+
+CMake:
+  - '**/CMakeLists.txt'
+  - '**/cmake/**'
+  
+cuDF (Java):
+  - 'java/**'
+
+ci:
```
Comment (mike-wendt) [inline .github/labeler.yml:19]: FYI this should be `gpuCI` to match the [label](https://github.com/rapidsai/cudf/labels?q=gpuci)

File: .github/labeler.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# https://github.com/actions/labeler#common-examples
+# Generated from https://github.com/rapidsai/cudf/blob/main/.github/CODEOWNERS
+# Labels culled from https://github.com/rapidsai/cudf/labels
+
+cuDF (Python):
+  - 'python/**'
+  - 'notebooks/**'
+  
+cpp:
```
Comment (mike-wendt) [inline .github/labeler.yml:9]: @kkraus14 should this label be `libcudf` or `libcudf++`? Or do you prefer a new label here of `cpp`? Just trying to match to an [existing label](https://github.com/rapidsai/cudf/labels) and `cpp` is not one currently

**Review by mike-wendt — CHANGES_REQUESTED** [review]

File: .github/labeler.yml (hunk 8 above)
Comment (kkraus14) [reply .github/labeler.yml:9]: `libcudf` if we want to match an existing label

**Review by mike-wendt — APPROVED** [review]

**Comment by mike-wendt** [comment]
rerun tests

**Comment by harrism** [comment]
Is this ready to merge once CI passes? @jolorunyomi @mike-wendt

**Comment by mike-wendt** [comment]
with `[skip-ci]` this will just stay pending - there is no way to change the statuses for this commit so we need an updated commit

File: .github/labeler.yml (hunk 8 above)
Comment (mike-wendt) [inline .github/labeler.yml]:
```suggestion
libcudf:
```

**Comment by harrism** [comment]
> with [skip-ci] this will just stay pending - there is no way to change the statuses for this commit so we need an updated commit

That's fine, can you please take ownership of getting this PR merged? I don't know whether it is ready.

File: .github/labeler.yml [hunk 9]
```diff
@@ -1,6 +1,4 @@
-# https://github.com/actions/labeler#common-examples
-# Generated from https://github.com/rapidsai/cudf/blob/main/.github/CODEOWNERS
-# Labels culled from https://github.com/rapidsai/cudf/labels
+# Documentation for conifg - https://github.com/actions/labeler#common-examples
```
Comment (harrism) [inline .github/labeler.yml:1]:
```suggestion
# Documentation for config - https://github.com/actions/labeler#common-examples
```

**Comment by mike-wendt** [comment]
This should pass shortly and can be merged with the bot command as it will pass all checks in a few mins

**Comment by mike-wendt** [comment]
@gpucibot merge

**Pull request merged**